modules/vsphere-problem-detector-metrics.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * storage/vsphere-problem-detector-operator.adoc
+
+:operator-name: vSphere Problem Detector Operator
+
+[id="vsphere-problem-detector-operator-metrics_{context}"]
+= Metrics for the vSphere problem detector
+
+The {operator-name} exposes the following metrics for use by the Prometheus-based {product-title} cluster monitoring stack.
+
+.Metrics exposed by the {operator-name}
+[cols="2a,8a",options="header"]
+|===
+|Name |Description
+
+|`vsphere_cluster_check_total`
+|Number of vSphere cluster-level checks that were performed by the problem detector. This count includes both successes and failures.
+
+|`vsphere_cluster_check_errors`
+|Number of failed vSphere cluster-level checks that were performed by the problem detector.
+
+|`vsphere_node_check_total`
+|Number of vSphere node-level checks that were performed by the problem detector. This count includes both successes and failures.
+
+|`vsphere_node_check_errors`
+|Number of failed vSphere node-level checks that were performed by the problem detector.
+
+|===
+
+////
+Node-level checks with vsphere_node_check_total:
+  Each time the detector runs, the detector increments the count
+  for each of the following checks:
+
+* CheckNodeDiskUUID
+* CollectNodeESXiVersion
+* CollectNodeHWVersion
+* CheckNodeProviderID
+
+Cluster-level checks with vsphere_cluster_check_errors:
+
+* CheckStorageClasses
+  * Identifies storage classes with long names that cannot
+    pass the 255 char length after applying the /var/... pattern
+    that is used with dynamic provisioning.
+  * Identifies storage classes that reference a storagePolicyName
+    that does not exit, or references more than one storage policy.
+* CheckPVs
+  * Identifies PVs that are created with very long volume paths.
+
+Cluster-level check totals with vsphere_cluster_check_total:
+  Each time the detector runs, the detector increments the count
+  for each of the following checks:
+CheckDefaultDatastore
+CheckFolderPermissions
+CheckPVs
+CheckStorageClasses
+CheckTaskPermissions
+ClusterInfo
+
+////
+
+// Tidy up
+:operator-name:
```

Comment:
You could add an 'Additional resources' section under this module (in the assembly file directly) that has a cross reference to **Monitoring** -> **Understanding the monitoring stack**.